runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -742,6 +749,30 @@ private void toggleSidebar()
 		}
 	}
 
+	private void togglePluginPanel()
+	{
+		// Toggle sidebar open
+		boolean isPanelOpen = sidebarOpen;
+		sidebarOpen = !sidebarOpen;
+
+		if (isPanelOpen)
+		{
+			contract();
+		}
+		else
+		{
+			// Try to restore last panel
+			expand(currentNavButton);
+
+			// Add the plugin toolbar if it was previously closed by toggleSidebar
+			if (!container.isAncestorOf(pluginToolbar))
+			{
+				container.add(pluginToolbar);
+				frame.expandBy(pluginToolbar.getWidth());
+			}
+		}
```

Comment:
Per Adam's comment, this causes the frame to expand by the plugin toolbar's width if you collapse the panel (F+12) then expand the sidebar (F+11). From what I can tell, that's partially because of the `sidebarOpen = !sidebarOpen;` state mutation which happens here.

Anyway, I think this whole method body can safely be replaced by the following:

```java
final boolean pluginPanelOpen = pluginPanel != null;

currentButton.setSelected(!pluginPanelOpen);

if (pluginPanelOpen)
{
	contract();
}
else
{
	expand(currentNavButton);
}
```

Reply:
I neglected to suggest a null check for `currentButton`, since that can be nulled when clicking a plugin panel open and closed again. I've amended your commit to add this.